pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentTopicTest.java

```diff
@@ -1868,14 +1868,8 @@ public void testCompactionTriggeredAfterThresholdFirstInvocation() throws Except
         Compactor compactor = pulsar.getCompactor();
         doReturn(compactPromise).when(compactor).compact(anyString());
 
-        Policies policies = new Policies();
-        policies.compaction_threshold = 1L;
-
-        NamespaceResources nsr = pulsar.getPulsarResources().getNamespaceResources();
-        NamespaceName ns = TopicName.get(successTopicName).getNamespaceObject();
-        doReturn(Optional.of(policies)).when(nsr).getPolicies(ns);
-
         PersistentTopic topic = new PersistentTopic(successTopicName, ledgerMock, brokerService);
+        topic.topicPolicies.getCompactionThreshold().updateNamespaceValue(1L);
```

Comment:
This does not cover the init procedure of `topicPolicies.getCompactionThreshold()`.

Reply:
You can use `topic.initialize()` to trigger the init.